en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/troubleshooting/debug-memory.md

```diff
@@ -0,0 +1,39 @@
+# Taking Memory Dumps
+
+As with any software product, there can be instances where WSO2 Choreo Connect cluster fails due to a memory resource exhaustion. The heap dumps will always point you towards the cause of the memory leak. Therefore, it is important to be able to retrieve heap dumps from an environment at the point when an error occurs. This will avoid the necessity of reproducing the exact issue again (specially, in the case of production issues). A resource exhaustion can happen for two reasons:
+
+- Due to a bug in the system.
+- An actual limitation of resources based on low configuration values.
+
+Following sections will guide you on how to take heap dumps from each component of WSO2 Choreo Connect.
+
+## Taking a heap dump of Adapter
+
+Adapter is implemented in Go lang, therefore it exposes standard `pprof` ([GO pprof pkg](https://pkg.go.dev/net/http/pprof)) based go profiling endpoints in `localhost`.
+Follow below steps to take a heap dump of adapter.
+
+1. Log into the adapter container or port-forward the port `6060` from adapter to localhost.
```

Comment:
```suggestion
1. Log into the Adapter container or port-forward the port `6060` from adapter to localhost.
```
Can we add some explaination of "port-forwarding" here and mention more about the concept of logging into the Adapter?

Reply:
port-forward is a standard kubernetes command. Added bit more context to the sentence.